tests/framework/e2e/e2e.go

```diff
@@ -91,7 +91,24 @@ func (e e2eRunner) NewCluster(ctx context.Context, t testing.TB, opts ...config.
 	if err != nil {
 		t.Fatalf("could not start etcd integrationCluster: %s", err)
 	}
-	return &e2eCluster{t, *epc}
+	clus := &e2eCluster{t, *epc}
+	if cfg.WithLearner {
```

Comment:
This is not a proper implementation of learner e2e. We should have a dedicated server that is in learner mode. We might need to modify tests so that they also do some validation on learner, for example check if it's up to date with main cluster.